### PR TITLE
Make compressed instructions work on all tests

### PR DESCRIPTION
**Pull request by donofrio**
To provide better compatibility with the RV toolchain all RV64 tests now build with RV64GC. To get all tests to pass two changes were required:

1) The prefetcher was not always properly accounting for 16-bit instructions. This was a single condition added to the IsAvail() function. I also removed some unnecessary recursion in the Fill() function.

2) Compressed instructions had their source and destination registers modified at execution time. This causes issues with Dependency check and LS queue as the registers used to Set/Clear are off by 8, or sometimes not set at all. All these modifications that were previously done at execute time are now incorporated into the Decode() function. This means that the RevInst has the correct source and destination registers after decode

### CODE REVIEW

**Comment by jleidel** [comment]
Currently failing: 
```
The following tests FAILED:
	  7 - TEST_BIG_LOOP (Failed)
	 18 - TEST_MEMSET_2 (Failed)
	 53 - fcvt_w (Failed)
	 55 - fma (Failed)
```

**Comment by donofrio** [comment]
All tests are now passing! Many thanks to all - especially @leekillough for their help getting this across the line

**Review by rkabrick — COMMENTED** [review]
Nothing earth shattering in my comments. Take it or leave it.

Get this MERGED

File: scripts/spikeCheck.py [hunk 1]
```diff
@@ -39,7 +39,7 @@
 for line in asmLines:
   splitLine = line.strip().split(":")
   PC = splitLine[0]
-  if "<_start>:" in line:
+  if "<main>:" in line:
```
[the remark this answers is not in the document]
Comment (donofrio) [reply scripts/spikeCheck.py:42]: ah - yes, that is a good idea, I'll fix that in a separate commit

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -93,7 +93,7 @@ add_test(NAME TEST_BIG_LOOP COMMAND run_big_loop.sh WORKING_DIRECTORY "${CMAKE_C
 set_tests_properties(TEST_BIG_LOOP
   PROPERTIES
     ENVIRONMENT "RVCC=${RVCC}"
-    TIMEOUT 100
+    TIMEOUT 200
```
[the remark this answers is not in the document]
Comment (donofrio) [reply test/CMakeLists.txt:96]: nope - removing

**Pull request merged**